drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/commands/actualcommands/AbstractScenarioSimulationCommand.java

```diff
@@ -181,7 +181,9 @@ protected ScenarioGridColumn getScenarioGridColumnLocal(ScenarioSimulationBuilde
     protected CommandResult<ScenarioSimulationViolation> commonExecution(ScenarioSimulationContext context) {
         context.getScenarioGridPanel().onResize();
         context.getScenarioGridPanel().select();
-        context.getScenarioGridPanel().setFocus(true);
+        if (undoable) {
```

Comment:
@jomarko 
Why this should be conditional ?

Reply:
@gitgabrio I tried to explain in PR description. The purpose of the `context.getScenarioGridPanel().setFocus(true);` at this place to keep canvas focused, in other words keep canvas reactive on keyboard shortcuts  after column was added, row deleted etc. All these actions are **undoable**, they have some UI effect. The issue is that `AbstractScenarioSimulationCommand` is extended also by commands without UI effect on canvas, for example `EnableRightPanelCmmand`. It means that user invokes context menu for some header cell, however at the same time right panel is focused (by `EnableRightPanelCommand`), the `setFocus(true)` is called, what causes destroying resources bound with canvas, the context menu is gone even before user spot it, or select something. Is it more clear now please?

Reply:
@jomarko
Scope of the undoable flag is to know of a given behavior (undo) is enabled on the specific instance. Now, there is not any kind of relationship between undoability and the kind of operation the command is meant ti execute. But you are changing the ui status only based on that flag. To try to explain differenti, a command could be undoable without ever interacting with ui, so why It should change the status of the panel focus. If the currently implemented commands are interacting directly with ui is only by chance, absolutely not by design.

Reply:
@gitgabrio I am sorry, I do not understand your answer, anyway, I will try to change the approach completely as I can not explain more or better this solution.